### PR TITLE
Class exists checks for settings classes

### PR DESCRIPTION
**Pull request by cliffordp**
just some `! class_exists()` wrapping

### CODE REVIEW

File: src/Settings.php [hunk 1]
```diff
@@ -4,229 +4,233 @@
...
-	 * The prefix for our settings keys.
-	 *
-	 * Gets set automatically from the Text Domain or can be set manually.
-	 * The prefix should not end with underscore `_`.
-	 *
-	 * @var string
-	 */
-	private $opts_prefix = '';
-
+if ( ! class_exists( Settings::class ) ) {
```
Comment (bordoni) [inline src/Settings.php:7]: You shouldn't use the class to check if the class exists.

File: src/Settings.php [hunk 2]
```diff
@@ -4,229 +4,233 @@
...
+
+		/**
+		 * Settings constructor.
+		 *
+		 * TODO: Update this entire class for your needs, or remove the entire `src` directory this file is in and do not load it in the main plugin file.
+		 *
+		 * @param string $opts_prefix
+		 */
+		public function __construct( $opts_prefix = '' ) {
+			$this->settings_helper = new Settings_Helper();
```
Comment (bordoni) [inline src/Settings.php:37]: Expose a public method to set this and to get this.

File: src/Settings.php [hunk 3]
```diff
@@ -4,229 +4,233 @@
...
+		 * Set the options prefix to be used for this extension's settings.
+		 *
+		 * Defaults to the text domain, converting hyphens to underscores.
+		 * Always has ends with a single underscore.
+		 *
+		 * @param string $opts_prefix
+		 */
+		private function set_options_prefix( $opts_prefix = '' ) {
+			if ( empty( $opts_prefix ) ) {
+				$opts_prefix = str_replace( '-', '_', PLUGIN_TEXT_DOMAIN );
```
Comment (bordoni) [inline src/Settings.php]: Lets avoid this being set like this, we should use something that would error out, so someone couldn't copy and pasta this without the replacement.

File: src/Settings.php [hunk 4]
```diff
@@ -4,229 +4,233 @@
...
-		$this->settings_helper->remove_field( 'embedGoogleMapsZoom', 'general' );
-	}
+		/**
+		 * Adds a new section of fields to Events > Settings > General tab, appearing after the "Map Settings" section and
+		 * before the "Miscellaneous Settings" section.
+		 */
+		public function add_settings() {
+			$fields = [
+				// TODO: Settings heading start. Remove this element if not needed. Also remove the corresponding `get_example_intro_text()` method below.
+				$this->opts_prefix . 'Example'   => [
```
Comment (bordoni) [inline src/Settings.php]: I would make a method to append the prefix, makes it error proof.

**Review by bordoni — CHANGES_REQUESTED** [review]
I also wonder how much o this could live in common.

File: src/Settings.php (hunk 1 above)
Comment (cliffordp) [reply src/Settings.php:7]: That's using the special _class_ constant:
* https://www.php.net/manual/en/language.oop5.constants.php
* 1min demo this code isn't a problem: https://share.getcloudapp.com/nOuNWQoY

File: src/Settings.php (hunk 2 above)
Comment (cliffordp) [reply src/Settings.php:37]: done

File: src/Settings.php (hunk 3 above)
Comment (cliffordp) [reply src/Settings.php]: done

File: src/Settings.php (hunk 4 above)
Comment (cliffordp) [reply src/Settings.php]: done

**Comment by cliffordp** [comment]
@bordoni thanks for helping make it better. please re-review.

**Review by bordoni — APPROVED** [review]

**Pull request merged**